odoo/addons/base/models/res_partner.py

```diff
@@ -160,9 +159,10 @@ def default_get(self, default_fields):
     parent_name = fields.Char(related='parent_id.name', readonly=True, string='Parent name')
     child_ids = fields.One2many('res.partner', 'parent_id', string='Contact', domain=[('active', '=', True)])  # force "active_test" domain to bypass _search() override
     ref = fields.Char(string='Reference', index=True)
-    lang = fields.Selection(_lang_get, string='Language',
+    lang = fields.Selection(_lang_get, string='Language Code', compute="_compute_lang", inverse="_inverse_lang", store=True, readonly=False,
                             help="All the emails and documents sent to this contact will be translated in this language.")
-    active_lang_count = fields.Integer(compute='_compute_active_lang_count')
+    lang_id = fields.Many2one('res.lang', string="Language", default=lambda self: self.env['res.lang']._lang_get_id(self.env.context.get('lang')),
+        domain=lambda self: [('code', 'in', self.env['res.lang']._get_enabled_lang_code())])
```

Comment:
Why getting by the lang code as you could directly do `id in get_available_lang().ids`
it implies that get_available lang returns res.lang records. 
To the get codes when needed simply do, get_available_lang.mapped('code')